cloudsim-plus/src/main/java/org/cloudbus/cloudsim/datacenters/DatacenterSimple.java

```diff
@@ -872,33 +798,19 @@ protected final void setLastProcessTime(final double lastProcessTime) {
     }
 
     @Override
-    public List<FileStorage> getStorageList() {
-        return Collections.unmodifiableList(storageList);
+    public DatacenterStorage getDatacenterStorage() {
+        return this.datacenterStorage;
     }
 
     /**
-     * Sets the list of storage devices of the Datacenter.
+     * Sets the storage of the Datacenter.
      *
-     * @param storageList the new storage list
+     * @param datacenterStorage the new storage object
      * @return
      */
     @Override
-    public final Datacenter setStorageList(final List<FileStorage> storageList) {
-        Objects.requireNonNull(storageList);
-        this.storageList = storageList;
-        setAllFilesOfAllStoragesToThisDatacenter();
-
-        return this;
-    }
-
-    /**
-     * Assigns all files of all storage devices to this Datacenter.
-     */
-    private void setAllFilesOfAllStoragesToThisDatacenter() {
-        storageList.stream()
-                .map(FileStorage::getFileList)
-                .flatMap(List::stream)
-                .forEach(file -> file.setDatacenter(this));
+    public void setDatacenterStorage(DatacenterStorage datacenterStorage) {
```

Comment:
When you call a method such as the setDatacenterStorage inside a constructor, it's good practice to make it final. That is to avoid the "overridable method call in constructor" issue.

Reply:
That's a very valuable tip thank you